### ISSUE TITLE
"Configuring incomplete, errors occurred!" error during install.

### ISSUE THREAD

**Issue by ashahba** [NervanaSystems/aeon#66]
I'm seeing this error while trying to install `aeon` based on README docs:

```
root@94adf47fd68f:/aeon# mkdir -p build && cd $_ && cmake .. && pip install .
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   filesystem
-- Could NOT find LATEX (missing:  LATEX_COMPILER) 
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
CMake Error at cmake/Modules/FindSphinx.cmake:32 (unset):
  unset called with an invalid second argument
Call Stack (most recent call first):
  CMakeLists.txt:207 (find_package)


CMake Warning at cmake/Modules/FindSphinx.cmake:33 (message):
  Sphinx: problem extracting version!
Call Stack (most recent call first):
  CMakeLists.txt:207 (find_package)


-- Could NOT find Sphinx (missing:  SPHINX_VERSION) 
Doxygen not found, skipping documentation
Sphinx not found, skipping documentation

===[ Build Summary ]=============================================================

  General:
    Version                    : 1.3.0
    System                     : Linux
    C++ Compiler               : /usr/bin/clang++
    Build Type                 : Debug

    AEON Library               : Yes
    AEON Service               : No
    AEON Python Interface      : Yes

  Dependencies:
    Boost                      : Yes (ver. 1.58)
    OpenCV                     : Yes (ver. 2.4.9.1)
    SoX                        : Yes (ver. 14.4.1)
    Curl                       : Yes (ver. 7.47.0)
    CppREST                    : Not Found
    Fabric                     : Not Found
    OpenSSL                    : Not Found

  Python:
    Interpreter                : /usr/bin/python3.5 (ver. 3.5.2)
    Libraries                  : /usr/lib/x86_64-linux-gnu/libpython3.5m.so (ver. 3.5.2)
    NumPy                      : /usr/local/lib/python3.5/dist-packages/numpy/core/include (ver. 1.13.1)

  Documentation:
    Doxygen                    : No
    LaTeX                      : No
    Sphinx                     : No
    Breathe                    : /usr/local/bin/breathe-apidoc (ver. 4.7.3)

Install:
    Installation path          : /usr/local

================================================================================

-- Configuring incomplete, errors occurred!
See also "/aeon/build/CMakeFiles/CMakeOutput.log".
See also "/aeon/build/CMakeFiles/CMakeError.log".
```

**Comment by ashahba** [comment]
Seems like we are missing `sphinx` in `requirement.txt` file.
I'll get a PR ready.

**Comment by sfraczek** [comment]
We have fixed this internally by modifying FindSphinx script to stop throwing error when it hasn't found Sphinx. After that update you will not have errors when running `cmake ..` without Sphinx and your pipeline should be fine then. Can you, for now, run `doc/install_prereqs.sh` before running `cmake ..` ? It should install Sphinx and satisfy cmake's requirements. We had a discussion on the inside recently that Sphinx is required only to build docs (like make html) and we have concluded that doc/install_prereqs.sh can be run beforehand to install it if needed for building docs. We also moved breathe from requirements.txt to that script because it has the same situation. 

What do you think?

**Comment by ashahba** [comment]
This sounds reasonable. When is the fix going to be merged into `master`?

**Comment by sfraczek** [comment]
Next week (11th).

**Comment by sfraczek** [comment]
Done. Released v1.3.1. 
Can you please verify if it solved your problem?

**Comment by leekwunfung817** [comment]
Config the CMake First
```

 cmake 
-DCMAKE_BUILD_TYPE=Release
-DCMAKE_INSTALL_PREFIX=/usr/local 
-DOPENCV_EXTRA_MODULES_PATH=../../<install_package_folder_name>/modules -DOPENCV_ENABLE_NONFREE=True ..
```
